gremlin-go/driver/authInfo.go

```diff
@@ -21,29 +21,40 @@ package gremlingo
 
 import "net/http"
 
-// AuthInfo is an option struct that allows authentication information to be specified.
-// Authentication can be provided via http.Header Header directly.
+// AuthInfoProvider is an interface that allows authentication information to be specified.
+type AuthInfoProvider interface {
+	GetHeader() http.Header
+	GetBasicAuth() (ok bool, username, password string)
+}
+
+// AuthInfo is an option struct that allows authentication information to be specified statically.
+// Authentication can be provided via http.Header directly.
 // Basic authentication can also be used via the BasicAuthInfo function.
 type AuthInfo struct {
 	Header   http.Header
 	Username string
 	Password string
 }
 
-// getHeader provides a safe way to get a header from the AuthInfo even if it is nil.
+var _ AuthInfoProvider = (*AuthInfo)(nil)
```

Comment:
This is so that we make sure `*AuthInfo` satisfies the `AuthInfoProvider` interface (compile-time check). More below with `*DynamicAuth`.